dask/dataframe/io/io.py

```diff
@@ -664,24 +669,27 @@ def from_delayed(
     if not dfs:
         dfs = [delayed(make_meta)(meta)]
 
-    name = prefix + "-" + tokenize(*dfs)
-    dsk = {}
-    if verify_meta:
-        for (i, df) in enumerate(dfs):
-            dsk[(name, i)] = (check_meta, df.key, meta, "from_delayed")
-    else:
-        for (i, df) in enumerate(dfs):
-            dsk[(name, i)] = df.key
-
     if divisions is None or divisions == "sorted":
         divs = [None] * (len(dfs) + 1)
     else:
         divs = tuple(divisions)
         if len(divs) != len(dfs) + 1:
             raise ValueError("divisions should be a tuple of len(dfs) + 1")
 
+    name = prefix + "-" + tokenize(*dfs)
+    layer = DataFrameIOLayer(
+        name=name,
+        columns=None,
+        inputs=BlockwiseDepDict(
+            {(i,): inp.key for i, inp in enumerate(dfs)},
+            produces_keys=True,
+        ),
+        io_func=(lambda x: check_meta(x, meta, "from_delayed"))
+        if verify_meta
+        else (lambda x: x),
```

Comment:
I agree that it is best to avoid lambdas - I added a simple (private) `_CheckMetaFunc` to avoid the need to serialize everything in `from_delayed`. I left this in io.py for now, but I suppose we could move this to utils.py if other APIs end up using it.  Let me know what you think

Reply:
Actually - I realized that using `lambda x: x` is not a problem, so I just used `functools.partial` for `check_meta(x, meta, "from_delayed")` (sorry for going in circles on this)